projects/admin/src/app/menu/menu-switch-library/menu-switch-library.component.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
+import { DialogService, MenuItem } from '@rero/ng-core';
+import { Observable } from 'rxjs';
+import { AppConfigService } from '../../service/app-config.service';
+import { LibrarySwitchMenuService } from './service/library-switch-menu.service';
+import { LibrarySwitchService } from './service/library-switch.service';
+
+@Component({
+  selector: 'admin-menu-swith-library',
+  templateUrl: './menu-switch-library.component.html'
+})
+export class MenuSwitchLibraryComponent {
+
+  /** Menu libraries switch */
+  get menuLibrariesSwitch() {
+    return this._librarySwitchMenuService.menu;
+  }
+
+  /** Menu is visible */
+  get isVisible() {
+    return this._librarySwitchMenuService.visible;
+  }
+
+  /**
+   * Constructor
+   * @param _appConfigService - AppConfigService
+   * @param _menuSwitchMenuService - LibrarySwitchMenuService
+   * @param _librarySwitchService - LibrarySwitchService
+   * @param _dialogService - DialogService
+   * @param _translateService - TranslateService
+   * @param _router - Router
+   */
+  constructor(
+    private _appConfigService: AppConfigService,
+    private _librarySwitchMenuService: LibrarySwitchMenuService,
+    private _librarySwitchService: LibrarySwitchService,
+    private _dialogService: DialogService,
+    private _translateService: TranslateService,
+    private _router: Router
+  ) { }
+
+  /**
+   * Event on menu click
+   * @param event - MenuItem
+   */
+  eventMenuClick(event: MenuItem) {
+    // TODO: Implement this with a guard (implement in ng-core and ui)
+    const url = this._router.url;
+    const urlParams = url.split('/').filter(n => n);
```

Comment:
perhaps you can use: https://angular.io/api/router/UrlSerializer

Reply:
done